base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
```

Comment:
- [ ] [Optional] You do not need the `super` method in this `initialize` method because the `Nameable` class has no initialize method. The mere fact that it inherits from Nameable, it will have the same attributes and methods as Nameable. 👍